Jenkinsfile-Release

```diff
@@ -0,0 +1,145 @@
+]
+
+final jiraVersionMappings = [
+  'nexus-repository-manager-single-instance': 'helm-nxrm',
+  'nexus-repository-manager-aws-resilient-single-instance': 'helm-nxrm-aws-resiliency'
+]
+
+final chartLocation = [
+  'nexus-repository-manager-single-instance': 'single-inst-oss-pro-kubernetes',
+  'nexus-repository-manager-aws-resilient-single-instance': 'aws-single-instance-resiliency'
+]
+
+properties([
+  parameters([
+    choice(
+      choices: ['', 'nexus-repository-manager-single-instance', 'nexus-repository-manager-aws-resilient-single-instance'],
+      name: 'chart',
+      description: 'Chart to deploy.',
+    ),
+    string(
+      name: 'appVersion',
+      description: 'Version of the application image, like "3.41.0"',
+    ),
+    string(
+      name: 'chartVersion',
+      description: '(Optional) Version of the Chart, like "41.0.0". If omitted, it will be calculated from the appVersion.',
+    ),
+  ])
+])
+
+final chartVersion = calculateChartVersion(params.chartVersion, params.appVersion)
+
+dockerizedBuildPipeline(
+  prepare: {
+    if (! params.chart) {
+      error('Chart parameter is required.')
+    }
+    if (! params.appVersion) {
+      error('The appVersion is required.')
+    }
+    githubStatusUpdate('pending')
+  },
+  buildAndTest: {
+    sonatypeZionGitConfig()
+    runSafely "git checkout ${gitBranch(env)}"
+    runSafely "./upgrade.sh ./${chartLocation[params.chart]} ${chartVersion} ${params.appVersion}"
+    runSafely './build.sh'
+    runSafely 'git add aws-single-instance-resiliency single-inst-oss-pro-kubernetes'
+  },
+  skipVulnerabilityScan: true,
+  archiveArtifacts: 'docs/*',
+  testResults: [],
+  deployCondition: { true },
+  deploy: {
+    runSafely 'git add docs'
+    runSafely "git commit -m 'Release Update for ${params.chart} ${chartVersion}'"
+
+    sshagent(credentials: [sonatypeZionCredentialsId()]) {
+      runSafely 'git push'
+    }
+  },
+  postDeploy: {
+    // Verify Index.yaml
+    String version = verifyIndexYamlAndTarFile(params.chart)
+
+    // Set Jira Fix Version
+    jira.versionPrefix = jiraVersionMappings[chart]
+    jiraSetFixVersion(jira, version)
+
+    // Create tags
+    String tagName = "${chart}-${version}"
+    runSafely "git tag -a ${tagName} -m 'Release Update: ${version}'"
+    sshagent(credentials: [sonatypeZionCredentialsId()]) {
+      runSafely "git push origin ${tagName}"
+    }
+  },
+  onSuccess: {
+    buildNotifications(currentBuild, env)
+  },
+  onFailure: {
+    buildNotifications(currentBuild, env)
+  }
+)
+
+String verifyIndexYamlAndTarFile(String chart) {
+  // Get current version
+  def indexFile = readYaml file: 'docs/index.yaml'
+  String version = indexFile.entries[chart][0].version
+
+  // Check tar file
+  String repo_url = 'https://sonatype.github.io/helm3-charts/'
+  verifyDownloadLinks(
+    urlParts: [repo_url, chart, '-', version],
+    urlSuffixes: ['.tgz'], retryCount: 2, retryDelay: 60
+  )
```

Comment:
this check will fail until the helm3-charts build has been updated to consume these charts and index them. this check will likely be better to do on in that project, depending how this is all accomplished.

Reply:
Removed: https://github.com/sonatype/nxrm3-helm-repository/pull/9/commits/2aa0f5dbf98e0757c61bda586bf64dbe6a0c570b